contracts/examples/ping-pong-egld/mandos/ping-pong-call-ping-second-user.scen.json

```diff
@@ -0,0 +1,76 @@
+{
+    "name": "call ping for a second user",
+    "steps": [
+        {
+            "step": "externalSteps",
+            "path": "ping-pong-call-ping.scen.json"
+        },
+        {
+            "step": "setState",
+            "accounts": {
+                "address:participant2": {
+                    "nonce": "0",
+                    "balance": "2,500,000,000,000",
+                    "storage": {}
+                }
+            },
+            "currentBlockInfo": {
+                "blockNonce": "781"
+            }
+        },
+        {
+            "step": "scCall",
+            "tx": {
+                "from": "address:participant2",
+                "to": "address:the_ping_pong_contract",
+                "value": "500,000,000,000",
+                "function": "ping",
+                "arguments": [],
+                "gasLimit": "100,000,000",
+                "gasPrice": "0"
+            },
+            "expect": {
+                "status": "",
+                "gas": "*",
+                "refund": "*"
+            }
+        },
+        {
+            "step": "checkState",
+            "accounts": {
+                "address:my_address": {
+                    "nonce": "1",
+                    "balance": "1,000,000",
+                    "storage": {}
+                },
+                "address:participant1": {
+                    "nonce": "1",
+                    "balance": "1,300,000,000,000",
+                    "storage": {}
+                },
+                "address:participant2": {
+                    "nonce": "1",
+                    "balance": "2,000,000,000,000",
+                    "storage": {}
+                },
+                "address:the_ping_pong_contract": {
+                    "nonce": "0",
+                    "balance": "1,000,000,000,000",
+                    "storage": {
```

Comment:
checkState steps are great, but they can be hard to keep up to date when refactoring the storage.
That is why we usually prefer to check the correctness via view functions/sc call outputs.

It's good that you did these kinds of tests, but you might want to avoid being too explicit with the storage for future ones...

... especially since I just requested some storage renames and a small redesign.

Reply:
Here I followed the pattern inside the mandos tests for the crowdfunding - and I see the rest of the smart contracts also test the final values by making use of checkState.
checkState is very specific and doesn't allow me to skip fields, hence it's a bit safer than the alternative (testing individual fields). However, I see how it could become a problem for more complex contracts in the future.
For this contract I think checkState is fine since I also implemented a map between (id) <-> (account, status) which I also want to fully test.
If the mapping functionality would have been a module and would have its own tests, if I'm re-using it here then of course, it won't make sense to test its internals.